source/adapters/hip/memory.cpp

```diff
@@ -275,9 +278,10 @@ UR_APIEXPORT ur_result_t UR_APICALL urMemGetInfo(ur_mem_handle_t hMemory,
         } else if constexpr (std::is_same_v<T, SurfaceMem>) {
           HIP_ARRAY3D_DESCRIPTOR ArrayDescriptor;
           UR_CHECK_ERROR(hipArray3DGetDescriptor(&ArrayDescriptor, Mem.Array));
-          const auto PixelSizeBytes =
-              GetHipFormatPixelSize(ArrayDescriptor.Format) *
-              ArrayDescriptor.NumChannels;
+          auto PixelSize = 0;
+          UR_RETURN_ON_FAILURE(
+              GetHipFormatPixelSize(ArrayDescriptor.Format, &PixelSize));
```

Comment:
Can we define the `PixelSize` as `int` type explicitly instead of using `auto`. Currently it is `auto`-deduced as an `int` due to the `0` on the right hand-side regardless but I don't see the need for `auto` in this specific instance. Also, I think ideally the `GetHipFormatPixelSize` function would accept `unsigned int*` as an argument for the size parameter.

Reply:
Absolutely, the original line was `auto`.  I changed the `GetHipFormatPixelSize` signature too. Cheers.